nimbus/src/dbcache.rs

```diff
@@ -85,6 +93,10 @@ impl DatabaseCache {
     }
 
     pub fn get_experiment_branch(&self, slug: &str) -> Result<Option<String>> {
-        self.get_data(|data| data.experiment_branches.get(slug).cloned())
+        self.get_data(|data| data.branches_by_exp.get(slug).cloned())
+    }
+
+    pub fn get_branch_slug_by_feature(&self, feature_id: &str) -> Result<Option<String>> {
+        self.get_data(|data| data.branches_by_feature.get(feature_id).cloned())
```

Comment:
@jhugman I like this in principle, but I'm wondering about the pragmatics of this.  Right now, https://github.com/mozilla-mobile/fenix/search?q=getExperimentBranch returns one invocation and two comments. I was under the impression that we just want to land this PR as soon as possible so that we get as much as we can (eg hopefully at least the SDK-190 edge case checking work).

Would doing the overloaded `get_experiment_branch` idea make SDK-187 (android-components) and SDK-188 (fenix) non-trivially easier?

Anyone else have thoughts here?

Reply:
My thought process: we either change the Rust here, and [some string changes in the Fenix codebase][1].

Or we could change Rust here, and write kotlin in android components and fenix.

The delays between updating Nimbus, then megazord, then Android Components, then Fenix is the same, but the sequencing is much less complicated.

[1]: https://github.com/mozilla-mobile/fenix/blob/0b764fd3e4eb40b3fc3bf569a9de70a75804a3f1/app/src/main/java/org/mozilla/fenix/experiments/Experiments.kt#L10

Reply:
I had to think on this a minute, but I do like what @jhugman is suggesting as is is less of a breaking change.  `get_experiment_branch(identifier)`, where `identifier` could be either an experiment slug or a feature-id and then internally checks both caches for the branch would keep the API the same on the surface and all we would need to do in Fenix then is to get the right version of Nimbus and change the strings, as James suggests, and we don't have to change the Android Component piece at all (at least if/when we implement the feature API).

Reply:
OK, implementation updated to use the strategy suggested by @jhugman.